src/bench/logging.cpp

```diff
@@ -17,32 +19,77 @@ static void Logging(benchmark::Bench& bench, const std::vector<const char*>& ext
     bench.run([&] { log(); });
 }
 
-static void LoggingYoThreadNames(benchmark::Bench& bench)
+// The test framework currently enables all categories by default, but in case
+// that changes, we set -debug=category in the benchmarks below when we expect a
+// category to be logged.
+
+static void LogPrintLevelWithThreadNames(benchmark::Bench& bench)
 {
-    Logging(bench, {"-logthreadnames=1"}, [] { LogPrintf("%s\n", "test"); });
+    Logging(bench, {"-logthreadnames=1", "-debug=validation"}, [] {
+        LogPrintLevel(BCLog::VALIDATION, BCLog::Level::Error, "%s\n", "test");
+    });
 }
-static void LoggingNoThreadNames(benchmark::Bench& bench)
+
+static void LogPrintLevelWithoutThreadNames(benchmark::Bench& bench)
 {
-    Logging(bench, {"-logthreadnames=0"}, [] { LogPrintf("%s\n", "test"); });
+    Logging(bench, {"-logthreadnames=0", "-debug=validation"}, [] {
+        LogPrintLevel(BCLog::VALIDATION, BCLog::Level::Error, "%s\n", "test");
+    });
+}
+
+static void LogPrintWithCategory(benchmark::Bench& bench)
+{
+    Logging(bench, {"-logthreadnames=0", "-debug=validation"}, [] {
+        LogPrint(BCLog::VALIDATION, "%s\n", "test");
+    });
+}
+
+static void LogPrintWithoutCategory(benchmark::Bench& bench)
+{
+    Logging(bench, {"-logthreadnames=0", "-debug=0"}, [] {
+        LogPrint(BCLog::VALIDATION, "%s\n", "test");
+    });
+}
+
+static void LogPrintfCategoryWithThreadNames(benchmark::Bench& bench)
```

Comment:
In 4db9dc8ac3baa9751702ca575965661e63384692:

Looks like you're introducing these benchmarks, with `BCLog::NET`, only to change them in a later commit (dfb40748911b9553a6e10c8376b032dac36bf938) to `BCLog::VALIDATION`? It's not obvious to me why the category needs changing, but this can probably be re-ordered so we aren't changing the same (newly introduced) lines of code, multiple times in the same PR.